classes/FaviconController.php

```diff
@@ -22,26 +28,36 @@ public function index()
 
         $finalPath = $inputPath = storage_path('app/media' . $settings->favicon);
 
-        if ($settings->favicon_16) {
+        return response()->file($finalPath, [
+            'Content-Type' => 'image/x-icon',
+        ]);
+    }
 
-            $destinationPath = storage_path('app/initbiz/seostorm/favicon/' . dirname($settings->favicon) . '/');
-            $finalPath = $outputPath = $destinationPath . basename($settings->favicon);
+    public function generateManifest()
+    {
+        $settings = Settings::instance();
+        $responseArray = [];
+        
+        if ($settings->webmanifest_enabled !== true) {
+            $controller = new Controller();
+            $controller->setStatusCode(404);
+            return $controller->run('/404');
+        }
 
-            if (!file_exists($outputPath)) {
-                if (
-                    !File::makeDirectory($destinationPath, 0777, true, true) &&
-                    !File::isDirectory($destinationPath)
-                ) {
-                    trigger_error(error_get_last(), E_USER_WARNING);
-                }
+        if ($settings->webmanifest_enabled === true) {
```

Comment:
will `if($settings->webmanifest_enabled)` be enough?

Reply:
yes, it will be